### DISCUSSION TITLE
"git branch" does not show orphan branch names. Why?

### DISCUSSION THREAD

**Discussion by Ricalleite** [ohmybash/oh-my-bash#396] [Q&A] [answered]
Hi guys!! If someone could help me answer me here, please! My problem is: after installing "oh my bash" on my machine, the command "git branch" doesn't work anymore. Worse: I uninstalled that but the problem persist... Any idea that what's going on here? Any solution?

----

**Edit** by @akinomyoga 

The OP has created a new repository and checked out a new branch with `git checkout -b test`. Then, the command `git branch` doesn't return any results. The question is what is going on here and what is the solution.

```bash
$ mkdir Projects
$ cd Projects
$ git init
$ git checkout -b test
$ git branch
$      <-- doesn't show any result
```

**Comment by akinomyoga** [comment]
> the command "git branch" doesn't work anymore.

How does it fail? Aren't there any error messages?

**Reply by Ricalleite** [reply]
Let me explain what I did. I created a directory called "Projects" and made "git init" on it. After that I did checkout and created another branch called "test" (git checkout -b test). So when I type "git branch" nothing returns! The cli still the same form... The only way that I can see which branch where I am is using "git status". So bizarre...

**Reply by akinomyoga** [reply]
This means that your new branch `test` actually has no commits, right?

Then, it is the default behavior of Git. The same happens with any system and is unrelated to Oh My Bash. The branch in git is actually implemented as *a tracking reference to a commit*, so if there are no commits for the branch, it means that the branch has not yet "*materialized*" (not sure about a more appropriate word). `git branch` subcommand doesn't seem to recognize the "branch" that hasn't yet *materialized*. You can also find a description of the behavior in the following answer to a Stack Overflow question.

- https://stackoverflow.com/a/47141310/4908404

**Reply by Ricalleite** [reply]
Great! Thanks for help and information about that behavior . Living and learning! : )

**Reply by akinomyoga** [reply]
Thanks. I made it a separate answer https://github.com/ohmybash/oh-my-bash/discussions/396#discussioncomment-4927533.

**Reply by akinomyoga** [reply]
I've also updated the question title to better describe the actual problem.

**Comment by akinomyoga** [comment, marked as answer]
An answer to this question (from https://github.com/ohmybash/oh-my-bash/discussions/396#discussioncomment-4927361):

This is the default behavior of Git. The branch in Git is implemented as _a tracking reference to a commit_. When there are no commits for the branch, it means that the branch has not yet "_materialized_". The `git branch` subcommand doesn't seem to recognize the "branch" that hasn't yet _materialized_. You can also find a description of the behavior in the following answer to a Stack Overflow question.

- https://stackoverflow.com/a/47141310/4908404
